sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/NetworkRequest.kt

```diff
@@ -54,7 +56,11 @@ internal abstract class NetworkRequest {
         }
 
         val url = buildURL()
-        val connection: HttpsURLConnection = url.openConnection() as HttpsURLConnection
+        val connection = if (URLUtil.isHttpsUrl(url.toString())) {
```

Comment:
We need to correctly determine our protocol for the connection. The SDK is not going to connect to a local klaviyo instance over HTTPS so I'm adding in HTTP connections